server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1081,6 +1094,14 @@ public void finalizeSnapshot(final ShardGenerations shardGenerations,
         final StepListener<RepositoryData> repoDataListener = new StepListener<>();
         getRepositoryData(repoDataListener);
         repoDataListener.whenComplete(existingRepositoryData -> {
+            final int existingSnapshotCount = existingRepositoryData.getSnapshotIds().size();
+            if (existingSnapshotCount >= maxSnapshotCount) {
+                listener.onFailure(new RepositoryException(metadata.name(), "Can not add another snapshot to this repository as it " +
+                        "already contains [" + existingSnapshotCount + "] snapshots and is configured to hold up to [" + maxSnapshotCount +
+                        "] snapshots only. Please increase repository setting [" + MAX_SNAPSHOTS_SETTING.getKey() +
```

Comment:
I think we should name the setting in the message (as done here) but think we shouldn't recommend increasing it if you hit the limit and should instead suggest deleting some snapshots and/or adjusting retention settings to bring the snapshot count below the limit.

Reply:
++ adjusted the exception message